### DISCUSSION TITLE
Issue: pattern is being ignored

### DISCUSSION THREAD

**Discussion by codeef** [eslint/eslint#18484] [Q&A]
### Environment

Node version: 20.12.0
npm version: 10.7.0
Local ESLint version: 9.2.0
Global ESLint version: 
Operating System: 

### What parser are you using?

@typescript-eslint/parser

### What did you do?

I moved from eslint v8 with .eslintrc file to v9 with flat config and the eslint.config.js file.
Since then, I get an error while running
`eslint --fix --max-warnings=0 src`

The error says:
`You are linting "src", but all of the files matching the glob pattern "src" are ignored.`
But I don't think I'm ignoring the src pattern anywhere

I have [this repo](https://github.com/codeef/eslint-try-0) where you can see all the relevant code to reproduce the issue

Can you please tell me what am I doing wrong and how can I fix it?

Thanks

### What did you expect to happen?

Expected eslint to run properly with the new flat config

### What actually happened?

Can't run `eslint --fix` and eslint also not working with VSCode IDE

### Link to Minimal Reproducible Example

https://github.com/codeef/eslint-try-0/tree/master

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by aladdin-add** [comment]
in eslint v9, the flat config has been the default: https://eslint.org/blog/2024/04/eslint-v9.0.0-released/#flat-config-is-now-the-default-and-has-some-changes

you can see the docs to upgrade the config: https://eslint.org/docs/latest/use/configure/migration-guide
or use `ESLINT_USE_FLAT_CONFIG=false` to use the deprecated config: https://eslint.org/docs/latest/use/configure/configuration-files-deprecated

**Reply by sahin52** [reply]
I'm using flat config with eslint v8.57.1 but I get this warning with this command:  
`npx eslint "src/**/*.ts" "src/**/*.tsx"  --config .\eslint.config.mjs`

**Reply by ShadSterling** [reply]
That page says "By default, eslintrc files lint all files", so why is it ignoring "src"?
